File: web/models/api-key.ts
// Legacy single-key types (backward compat)
export type ApiKeyInfo = {
  masked_key: string
  created_at: string
  updated_at: string
}

export type ApiKeyFull = {
  key_value: string
}

// Multi-key management types
export type ApiKeyItem = {
  id: number
  name: string
  description: string | null
  masked_key: string
  scopes: string[]
  status: string
  created_at: string
  updated_at: string
}

export type ApiKeyCreatePayload = {
  name: string
  scopes: string[]
  description?: string
}

export type ApiKeyCreateResponse = ApiKeyItem & {
  key_value: string
}

export type ApiKeyListResponse = {
  items: ApiKeyItem[]
  total: number
  page: number
  per_page: number
  pages: number
}

File: server/app/schemas/api_key.py
"""
API Key schemas
"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.base import PaginatedResponse


# --- Legacy single-key schemas (backward compat) ---

class ApiKeyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    masked_key: str
    created_at: datetime | None = None
    updated_at: datetime | None = None


class ApiKeyFullResponse(BaseModel):
    key_value: str


# --- Multi-key management schemas ---

class ApiKeyCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=128)
    scopes: list[str] = Field(default=["chat"])
    description: str | None = None


class ApiKeyItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: str | None = None
    masked_key: str
    scopes: list[str]
    status: str
    created_at: datetime | None = None
    updated_at: datetime | None = None


class ApiKeyCreateResponse(ApiKeyItemResponse):
    """Returned on create/rotate — includes one-time full key."""
    key_value: str


class ApiKeyListResponse(PaginatedResponse):
    items: list[ApiKeyItemResponse]
